source/backend/routes/users.js

```diff
@@ -86,4 +88,45 @@ router.delete('/:savedRecipeId', async (req, res) => {
   }
 });
 
+/* POST /auth/signup */
+/* Creates new user in database with username, password and email -> unique user/email 
+required */
+router.post('/auth/signup', async (req, res) => {
+  try{
+    const { username } = req.username;
+    const { password } = req.password;
+    const hashedPwd = await bcrypt.hash(password, 10);
+    const { email } = req.email;
+
+    /* check username and email are unique */
```

Comment:
A better solution for this is to do a query with an and clause instead of getByUsername and getByEmail separately.
1. make a new function called `getByUsernameOrEmail(username, email)` that basically queries any row that has the same username or email.
2. if the array returned by that is not length of 0, return a 401 forbidden request

```
const existingUsers = await usersModel.getByUsernameOrEmail(username, email);
if (existingUsers.length !== 0) {
  return res.status(401).json({msg: 'Signup failed, Username or email already exisits'});
}
```

In the industry, to protect user information, we choose to omit telling the user if username or email is being used per https://security.stackexchange.com/questions/17816/username-and-or-password-invalid-why-do-websites-show-this-kind-of-message-i

Even though the above article is referring to password and username, same principle can be applied here.